website/content/docs/api-clients/api/pagination.mdx

```diff
@@ -0,0 +1,129 @@
+---
+layout: docs
+page_title: API list pagination
+description: Learn how the API pagination and cache works in Boundary to prevent system resources from being overwhlemed and to help you find resources.
+---
+
+# API list pagination
+
+Boundary uses an API pagination feature to support searching and filtering large lists of results without overwhelming your system resources.
+
+## Introduction
+
+When you perform a `list` operation on a resource, the API request returns the first 1000 results, by default.
+You can request the next page of results by repeating the `list` command and including a parameter with a token provided by Boundary.
+
+The results of the `list` operation are cached locally to help protect your system resources.
+Maintaining a local cache of results for a frequently used `list` operation can also help to reduce server cost.
+You can refresh the local results of your initial `list` operation as many times as you want for up to 30 days.
+
+## Initial pagination
+
+When you run a `list` operation on a resource, the initial results are sorted in descending order by creation time and public ID, by default.
+Boundary returns the maximum number of responses along with the top-level fields `response_type` and `list_token`.
+If the `response_type` is `delta`, it means there are more items available in the list.
+
+To request the next page of results, run the `list` operation again and include the `list_token` that Boundary provided as a parameter.
+Boundary returns the next page of results with new `response_type` and `list_token` values.
+
+![API pagination workflow](/img/api-pagination/pagination-workflow.png)
+
+You can continue viewing new pages of results using the `list_token` values until the `response_type` reads `complete`.
+
+## Refresh the local cache
+
+When the `response_type` reads `complete`, you have viewed all the results that were available when you ran the initial `list` operation.
+You can now refresh the cached list of results to return any resources that were updated since you started viewing the list responses.
+
+To refresh the cached list results, make another `list` request using the `list_token` from the previous response.
+Boundary returns a `response_type` and `list_token` as part of the response, similarly to when you ran the initial operation.
+If the `response_type` is `delta`, it means there are more items available in the list.
+You can use the `list_token` as a parameter to continue viewing new pages of results until the `response_type` reads `complete`.
+
+Boundary also includes the top-level response `deleted_ids` as part of the refresh operation.
+The `deleted_ids` response contains the IDs of any resources that were deleted since the initial `list` operation.
+Deleted resources are included in the results of the list for 30 days after they are deleted.
+Resources that have been deleted include the suffix `_deleted`.
```

Comment:
There appears to be some confusion here. The deleted resources returned are always relative to the start of the previous "pagination phase". E.g. for the first "refresh" phase it will, on the first page, include the IDs of any resources deleted since the start of the pagination. To illustrate:

```
1. Start paginating
2. A resource is deleted
3. Finish paginating
4. Start a "refresh phase". This response will include the ID of the resource deleted in 2.
5. Another resource is deleted.
6. Finish paginating the refresh phase.
7. Start another refresh phase. This response will include the ID of the resource deleted in 5.
```

Does that make sense?

Reply:
Makes sense, thank you! I changed this section to read:

> Boundary also includes the top-level response `removed_ids` as part of the refresh operation.
> The `removed_ids` response contains the IDs of any resources that were deleted since you began paginating the results of the previous `list` operation.
> Any resources that have been deleted are included on the first page of results.